frontend/lib/uast/post-parse-checks.cpp

```diff
@@ -1124,6 +1125,10 @@ void Visitor::checkReservedSymbolName(const NamedDecl* node) {
   } else if (isNameReservedType(name)) {
     error(node, "attempt to redefine reserved type '%s'.", name.c_str());
   }
+
+  if(strchr(name.c_str(), '$') != nullptr) {
+    warn(node, "Using $ in identifiers is deprecated; rename this to not use a $.");
```

Comment:
I'm not sure on the style recommendation here, but I typically wrap things I'm calling out in the warnings/errors with single quotes. Here I would wrap the `$`s, but that may just be me.

Reply:
There's no style guide for error message AFAIK, but I think quoting things like that makes sense.

I've updated it here: [`41566ff` (#23166)](https://github.com/chapel-lang/chapel/pull/23166/commits/41566ffe608f0b96ccc30f27bf590815826a6b1c)